modules/effects/src/provide_effects.ts

```diff
@@ -66,6 +69,13 @@ export function provideEffects(
 
   return makeEnvironmentProviders([
     effectsClasses,
+    EffectSources,
+    EffectsRunner,
+    Actions,
+    {
+      provide: EFFECTS_ERROR_HANDLER,
+      useFactory: () => defaultEffectsErrorHandler,
+    },
```

Comment:
Would it be still possible to use `provideEffects` at the child injector level? 🤔 We would need to add a check to ensure that these services are not provided multiple times.

Currently, we don't need to call `provideEffects` at the root level if there are no root effects. Even if we add a check to ensure that they're not provided multiple times, this change would introduce a breaking change, so using `provideEffects` at the root level would be required similar to `EffectsModule.forRoot`.

An option that we can consider is keeping the same behavior for `provideEffects` and introducing this change only for module-based apps.

Reply:
Yea, good point. I don't know what the behavior is for standalone APIs with module federation, or if we'll have the same issue there.